src/ops/factory.rs

```diff
@@ -0,0 +1,901 @@
+use crate::array::ArrayElement;
+use crate::error::DataStoreError;
+use crate::{array::Array, stream::StreamOrDevice};
+use mlx_macros::default_device;
+use num_traits::NumCast;
+
+impl Array {
+    /// Construct an array of zeros.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::{Array, StreamOrDevice};
+    /// Array::zeros_device::<f32>(&[5, 10], StreamOrDevice::default());
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - shape: Desired shape
+    /// - stream: Stream or device to evaluate on
+    #[default_device]
+    pub fn zeros_device<T: ArrayElement>(shape: &[i32], stream: StreamOrDevice) -> Array {
+        // TODO: Can we make use of full() here?
+        Self::try_zeros_device::<T>(shape, stream).unwrap()
+    }
+
+    /// Construct an array of zeros without validating shape.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::{Array, StreamOrDevice};
+    /// unsafe { Array::zeros_device_unchecked::<f32>(&[5, 10], StreamOrDevice::default()) };
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - shape: Desired shape
+    /// - stream: Stream or device to evaluate on
+    #[default_device]
+    pub unsafe fn zeros_device_unchecked<T: ArrayElement>(
```

Comment:
`clippy` seems to complain about not documenting why this is unsafe. It wants to see something like. But this is not a big deal.

```rust
/// # Safety
/// 
/// <placeholder>
```